ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java

```diff
@@ -0,0 +1,29 @@
+package com.spectralogic.ds3autogen.c.converters;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3autogen.api.models.Ds3Type;
+import com.spectralogic.ds3autogen.c.helpers.StructHelper;
+import com.spectralogic.ds3autogen.c.models.Struct;
+import com.spectralogic.ds3autogen.c.models.StructMember;
+
+import java.text.ParseException;
+
+public class StructConverter {
+    private final Ds3Type ds3Type;
+
+    private StructConverter(final Ds3Type ds3Type) {
+        this.ds3Type = ds3Type;
+    }
+
+    private Struct convert() throws ParseException {
+        final ImmutableList<StructMember> variablesList = StructHelper.convertDs3Elements(this.ds3Type.getElements());
+        return new Struct(
+                this.ds3Type.getName(),
+                variablesList);
+    }
+
+    public static Struct toStruct(final Ds3Type ds3Type) throws ParseException {
+        final StructConverter converter = new StructConverter(ds3Type);
```

Comment:
Why are you creating an instance of the class to perform the conversion?  Why not just use a static method.  This seems like it's creating an unnecessary object.